examples/vpc-peering-common/main.tf

```
@@ -0,0 +1,200 @@
+  files = merge(
+    { for k, v in var.vmseries : "files/${k}/config/bootstrap.xml" => "${k}/config/bootstrap.xml" },
+    { for k, v in var.vmseries : "files/${k}/config/init-cfg.txt" => "${k}/config/init-cfg.txt" },
+  )
+}
+
+module "vpc" {
+  source = "../../modules/vpc"
+
+  networks = { for k, v in var.networks : k => merge(v, {
+    name            = "${var.name_prefix}${v.name}"
+    subnetwork_name = "${var.name_prefix}${v.subnetwork_name}"
+    })
+  }
+
+}
+
+resource "google_compute_route" "this" {
+
+  for_each = var.routes
+
+  name         = "${var.name_prefix}${each.value.name}"
+  dest_range   = each.value.destination_range
+  network      = module.vpc.networks["${var.name_prefix}${each.value.network}"].self_link
+  next_hop_ilb = module.lb_internal[each.value.lb_internal_key].forwarding_rule
+  priority     = 100
+}
+
+module "vpc_peering" {
+  source = "../../modules/vpc-peering"
+
+  for_each = var.vpc_peerings
+
+  local_network = module.vpc.networks["${var.name_prefix}${each.value.local_network}"].id
+  peer_network  = module.vpc.networks["${var.name_prefix}${each.value.peer_network}"].id
+
+  local_export_custom_routes                = each.value.local_export_custom_routes
+  local_import_custom_routes                = each.value.local_import_custom_routes
+  local_export_subnet_routes_with_public_ip = each.value.local_export_subnet_routes_with_public_ip
+  local_import_subnet_routes_with_public_ip = each.value.local_import_subnet_routes_with_public_ip
+
+  peer_export_custom_routes                = each.value.peer_export_custom_routes
+  peer_import_custom_routes                = each.value.peer_import_custom_routes
+  peer_export_subnet_routes_with_public_ip = each.value.peer_export_subnet_routes_with_public_ip
+  peer_import_subnet_routes_with_public_ip = each.value.peer_import_subnet_routes_with_public_ip
+}
+
+module "vmseries" {
+  source = "../../modules/vmseries"
+
+  for_each = var.vmseries
+
+  name                  = "${var.name_prefix}${each.value.name}"
+  zone                  = each.value.zone
+  ssh_keys              = var.vmseries_common.ssh_keys
+  vmseries_image        = var.vmseries_common.vmseries_image
+  machine_type          = each.value.machine_type
+  min_cpu_platform      = try(each.value.min_cpu_platform, "Intel Cascade Lake")
+  tags                  = try(each.value.tags, [])
+  service_account       = module.iam_service_account[each.value.service_account].email
+  scopes                = try(each.value.scopes, [])
+  create_instance_group = true
+
+  bootstrap_options = try(merge({ vmseries-bootstrap-gce-storagebucket = "${module.bootstrap[each.value.bootstrap-bucket-key].bucket_name}/${each.key}/" }, var.vmseries_common.bootstrap_options), merge(
+    try(each.value.bootstrap_options, {}),
+    try(var.vmseries_common.bootstrap_options, {})
+  ))
+
+  named_ports = try([for v in each.value.named_ports :
+    {
+      name = v.name
+      port = v.port
+  }], [])
```

```suggestion
  named_ports = try(each.value.named_ports, [])
```
Any reason to rewrite a list into the same list?

---

Re-written based on suggestion.